ngraph/core/src/op/dft.cpp

```diff
@@ -0,0 +1,260 @@
+        }
+    }
+
+    PartialShape input_shape = PartialShape(get_input_partial_shape(0));
+    if (input_shape.rank().is_static())
+    {
+        const auto input_rank = input_shape.rank().get_length();
+        NODE_VALIDATION_CHECK(this,
+                              input_rank >= 2,
+                              "The input rank must be greater or equal to 2. Got input rank: ",
+                              input_rank);
+
+        auto last_dim_with_two = input_shape[input_rank - 1] & Dimension(2);
+        NODE_VALIDATION_CHECK(this,
+                              !last_dim_with_two.get_interval().empty(),
+                              "The last dimension of input data must be 2. Got: ",
+                              input_shape[input_rank - 1]);
+    }
+
+    PartialShape axes_shape = PartialShape(get_input_partial_shape(1));
+    if (axes_shape.rank().is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              axes_shape.rank().get_length() == 1,
+                              "DFT axes input must be 1D tensor. Got axes input rank: ",
+                              axes_shape.rank().get_length());
+    }
+
+    if (input_shape.rank().is_static() && axes_shape.is_static())
+    {
+        const auto input_rank = input_shape.rank().get_length();
+        NODE_VALIDATION_CHECK(this,
+                              input_rank >= axes_shape.to_shape()[0] + 1,
+                              "The input rank must be greater than number of DFT axes. Got "
+                              "input rank: ",
+                              input_rank,
+                              ", number of axes: ",
+                              axes_shape.to_shape()[0]);
+    }
+
+    if (input_shape.rank().is_static() && is_type<op::Constant>(input_value(1).get_node()))
+    {
+        const auto input_rank = input_shape.rank().get_length();
+        const auto& const_axes = get_constant_from_source(input_value(1));
+        auto axes = const_axes->cast_vector<int64_t>();
+
+        for (int64_t& axis : axes)
+        {
+            if (axis < 0)
+            {
+                axis += input_rank - 1;
+            }
+        }
+
+        AxisVector axes_vector;
+        AxisSet axes_set;
+        for (const int64_t axis : axes)
+        {
+            axes_vector.push_back(static_cast<size_t>(axis));
+            axes_set.insert(static_cast<size_t>(axis));
+        }
+
+        NODE_VALIDATION_CHECK(
+            this, axes.size() == axes_set.size(), "DFT axes must be unique. Got: ", axes_vector);
+
+        NODE_VALIDATION_CHECK(this,
+                              std::find(axes.begin(), axes.end(), input_rank - 1) == axes.end(),
+                              "DFT axes cannot contain the last axis. Got axes: ",
+                              axes_vector);
+    }
+
+    if (num_of_inputs == 3 && axes_shape.is_static() && get_input_partial_shape(2).is_static())
```

Comment:
can we merge two `if`s:
`88:    if (num_of_inputs == 3)` and `174:    if (num_of_inputs == 3 && axes_shape.is_static() && get_input_partial_shape(2).is_static())`?

Reply:
Done.